libs/recnet-api-model/src/lib/api/user.ts

```diff
@@ -42,12 +48,13 @@ export type PatchUserMeResponse = z.infer<typeof patchUserMeResponseSchema>;
 // POST /users/me
 export const postUserMeRequestSchema = userPreviewSchema
   .omit({
+    id: true,
     numFollowers: true,
   })
   .extend({
-    provider: z.enum(["FACEBOOK", "GOOGLE", "GITHUB"]),
-    providerId: z.string(),
+    email: z.string(),
     inviteCode: z.string(),
+    role: userRoleSchema,
```

Comment:
1. I would prefer having the email here, because if we want to support more providers, those providers might not include email in their jwt token.
2. The role was removed from the creation interface.